libs/tendermint/mempool/clist_mempool.go

```diff
@@ -305,6 +309,22 @@ func (mem *CListMempool) TxsWaitChan() <-chan struct{} {
 //
 // Safe for concurrent use by multiple goroutines.
 func (mem *CListMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo TxInfo) error {
+	mem.peersTxCountMtx.Lock()
```

Comment:
```suggestion
	mem.peersTxCountMtx.Lock()
	defer mem.peersTxCountMtx.Unlock()
```

Reply:
checktx has updateMtx.lock after this line. if defer unlock, it will be deadlock